pkg/sessions/dynamodb/dynamodb_store.go

```diff
@@ -0,0 +1,63 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/session"
+	dynamo "github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/sessions/persistence"
+	"time"
+)
+
+// SessionStore is an implementation of the persistence.Store
+// interface that stores sessions in redis
+type SessionStore struct {
+	Client Client
+}
+
+func NewDynamoDbSessionStore(opts *options.SessionOptions, cookieOpts *options.Cookie) (sessions.SessionStore, error) {
+	client, err := NewDynamoDbClient(opts.DynamoDb)
+	if err != nil {
+		return nil, fmt.Errorf("error constructing redis client: %v", err)
+	}
+
+	ss := &SessionStore{
+		Client: client,
+	}
+	return persistence.NewManager(ss, cookieOpts), nil
+}
+
+func NewDynamoDbClient(dynamodb options.DynamoDbStoreOptions) (Client, error) {
+
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	client := dynamo.New(sess)
+
+	c := newClient(client)
+
+	return *c, nil
+}
+
+func (s *SessionStore) Save(ctx context.Context, s2 string, bytes []byte, duration time.Duration) error {
```

Comment:
Rather than using `s2` it would be better to call this variable `key` as is done in the redis store